tests/statistics_test.py

```diff
@@ -0,0 +1,243 @@
+
+    @classmethod
+    def setUpClass(cls):
+        cls.rc = cls.create_rc()
+        cls.cluster = cls.create_cluster(cls.rc)
+        cls.member = cls.cluster.start_member()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.rc.exit()
+
+    def test_statistics_disabled_by_default(self):
+        client = HazelcastClient()
+        time.sleep(2 * self.DEFAULT_STATS_PERIOD)
+        response = self._get_client_stats_from_server()
+
+        self.assertTrue(response.success)
+        self.assertIsNone(response.result)
+        client.shutdown()
+
+    def test_statistics_disabled_with_wrong_value(self):
+        config = ClientConfig()
+        config.set_property(ClientProperties.STATISTICS_ENABLED.name, "truee")
+        config.set_property(ClientProperties.STATISTICS_PERIOD_SECONDS.name, self.STATS_PERIOD)
+        client = HazelcastClient(config)
+        time.sleep(2 * self.STATS_PERIOD)
+        response = self._get_client_stats_from_server()
+
+        self.assertTrue(response.success)
+        self.assertIsNone(response.result)
+        client.shutdown()
+
+    def test_statistics_enabled(self):
+        config = ClientConfig()
+        config.set_property(ClientProperties.STATISTICS_ENABLED.name, True)
+        client = HazelcastClient(config)
+
+        time.sleep(2 * self.DEFAULT_STATS_PERIOD)
+        self._wait_for_statistics_collection(2 * self.STATS_PERIOD)
+
+        client.shutdown()
+
+    def test_statistics_enabled_with_environment_variable(self):
+        environ = os.environ
+        environ[ClientProperties.STATISTICS_ENABLED.name] = "true"
+        environ[ClientProperties.STATISTICS_PERIOD_SECONDS.name] = str(self.STATS_PERIOD)
+
+        client = HazelcastClient()
+
+        time.sleep(2 * self.STATS_PERIOD)
+        self._wait_for_statistics_collection(2 * self.STATS_PERIOD)
+
+        os.unsetenv(ClientProperties.STATISTICS_ENABLED.name)
+        os.unsetenv(ClientProperties.STATISTICS_PERIOD_SECONDS.name)
+        client.shutdown()
+
+    def test_statistics_period(self):
+        config = ClientConfig()
+        config.set_property(ClientProperties.STATISTICS_ENABLED.name, True)
+        config.set_property(ClientProperties.STATISTICS_PERIOD_SECONDS.name, self.STATS_PERIOD)
+        client = HazelcastClient(config)
+
+        time.sleep(2 * self.STATS_PERIOD)
+        response1 = self._wait_for_statistics_collection(2 * self.STATS_PERIOD)
+
+        time.sleep(2 * self.STATS_PERIOD)
+        response2 = self._wait_for_statistics_collection(2 * self.STATS_PERIOD)
+
+        self.assertNotEqual(response1, response2)
+        client.shutdown()
+
+    def test_statistics_enabled_with_negative_period(self):
+        config = ClientConfig()
+        config.set_property(ClientProperties.STATISTICS_ENABLED.name, True)
+        config.set_property(ClientProperties.STATISTICS_PERIOD_SECONDS.name, -1 * self.STATS_PERIOD)
+        client = HazelcastClient(config)
+
+        time.sleep(2 * self.DEFAULT_STATS_PERIOD)
```

Comment:
is it possible to use assert eventually here instead of time sensitive test?

Reply:
Okay, I will increase the assertion timeout to 30 seconds from 2 * stats period